### ISSUE TITLE
[chart] Charts not clickable on Chrome

### ISSUE THREAD

**Issue by rajgoel** [rajgoel/reveal.js-plugins#20]
As reported in #5 charts are not clickable on Chrome. ~~Issue might be resolved if #19 is done.~~

**Comment by rajgoel** [comment]
Issue is probably a scaling related issue and caused by wrong mouse coordinates. Issue persists with Chart.js v2.3

**Comment by rajgoel** [comment]
Could be the same issue as https://github.com/chartjs/Chart.js/issues/1996 and similar to https://github.com/d3/d3/issues/2810

**Comment by alessandrogreco10** [comment]
Issue is still present even after update to Chart.min.js v2.5
wrong mouse coordinates in full screen mode on both chrome and safari

**Comment by morckx** [comment]
I've found a fix: 

[https://github.com/morckx/Chart.js/commit/449a3c0185894218ec0847702415acf2885dae92](https://github.com/morckx/Chart.js/commit/449a3c0185894218ec0847702415acf2885dae92)

and will make pull request.

**Comment by rajgoel** [comment]
Thanks a lot for your efforts! Haven't found time to test it yet, but here's a quick comment/question:

Would it also be possible to adjust  ```mouseX``` and ```mouseY``` similar to this [fix](https://github.com/mauriciopoppe/function-plot/issues/53#issuecomment-213981346) without explicitly checking for ```Reveal``` and calculating the scale by

```
scaleX = parentNode.offsetWidth  / parentNode.getBoundingClientRect().width;
scaleY = parentNode.offsetHeight  / parentNode.getBoundingClientRect().height;
```
where ```parentNode``` is the element containing the chart, i.e. Reveal.js. This might then be a general fix for the scaling issue.

**Comment by morckx** [comment]
Thanks for the suggestion, but this doesn't seem to work. If I determine the parentNode like this:

```
var revealRoot = canvas.parentNode;
while(revealRoot.parentNode != null && !revealRoot.classList.contains("reveal")) {
  revealRoot = revealRoot.parentNode;
}
```
and calculate scaleX and scaleY like this:
```
var revealRootBBox =  revealRoot.getBoundingClientRect();
var scaleX = revealRoot.offsetWidth  / revealRootBBox.width;
var scaleY = revealRoot.offsetHeight  / revealRootBBox.height;
console.log("scale:"+scale+" scaleX:"+scaleX+ " scaleY:"+scaleY);
```
scaleX and scaleY always seem to be 1

**Comment by rajgoel** [comment]
Unfortunately, #36 requires to modify Chart.js, thus the problem might reappear with future updates of Chart.js.

**Comment by rajgoel** [comment]
Will be fixed in [v2.8](https://github.com/chartjs/Chart.js/milestone/14) of chart.js
